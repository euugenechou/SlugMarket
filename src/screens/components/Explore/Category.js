// React imports 
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";

class Category extends Component {
  render() {
    return (
      <View style = { styles.container }>
        <View style = {{ flex: 2 }}>
          <Image
            source = { this.props.imageUri }
            style = { styles.image }
          />
        </View>
        <View style = { styles.text }>
          <Text style = {{ fontWeight: '600' }}> { this.props.name } </Text>
        </View>
      </View>
    );
  }
}

export default Category;

const styles = StyleSheet.create({
  text: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  container: {
    height: 120,
    width: 130,
    marginLeft: 20,
    borderRadius: 3,
    borderWidth: 1,
    borderColor: "#dddddd",
    shadowOffset: { width: 0, height: 0 },
    shadowColor: "black",
    shadowOpacity: 0.2,
    elevation: 1,
    backgroundColor: 'white'
  },
  image: {
    flex: 1,
    width: null,
    height: null,
    resizeMode: "cover",
    borderWidth: 0.1,
    borderRadius: 3,
    shadowOffset: { width: 0, height: 0 },
    shadowColor: "black",
    shadowOpacity: 0.2
  },
});
