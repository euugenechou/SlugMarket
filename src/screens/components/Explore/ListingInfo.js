/* React imports */
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { Button } from "react-native-elements";

class Listings extends Component {
  static navigationOptions = () => ({
    headerTintColor: "white",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      shadowColor: "transparent",
      borderBottomWidth: 0
    },
    headerBackground: (
      <Image
        style={StyleSheet.absoluteFill}
        source={require("../../../assets/textbooks.jpg")}
      />
    ),
  });

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.titleText}>
          {this.props.navigation.getParam("name")}
        </Text>
        <Text style={styles.text}>
          ${this.props.navigation.getParam("price")}
        </Text>
        <Text style={styles.text}>
          {this.props.navigation.getParam("category")}
        </Text>
        <Text style={styles.text}>
          {this.props.navigation.getParam("description")}
        </Text>
        <Button
          raised
          color="white"
          backgroundColor="teal"
          borderRadius={5}
          fontWeight="bold"
          onPress={() =>
            this.props.navigation.navigate("SellerDetails", {
              userId: this.props.navigation.getParam("userId"),
              seller: this.props.navigation.getParam("seller"),
              phoneNumber: this.props.navigation.getParam("phoneNumber"),
              email: this.props.navigation.getParam("email")
            })
          }
          title="View Seller Details"
          containerViewStyle={styles.buttonContainer}
        />
      </View>
    );
  }
}

export default Listings;

const styles = StyleSheet.create({
  text: {
    fontSize: 22,
    color: "black",
    fontWeight: "400",
    alignSelf: 'center',
    paddingBottom: 10,
    paddingHorizontal: 30,
  },
  titleText: {
    fontSize: 30,
    color: "black",
    fontWeight: "800",
    alignSelf: 'center',
    paddingBottom: 10,
    marginTop: 100
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white"
  },
  buttonContainer: {
    width: 300,
    alignSelf: 'center',
    paddingTop: 10
  }
});
