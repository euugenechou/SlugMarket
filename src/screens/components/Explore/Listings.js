// React imports
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";

class Listings extends Component {
  render() {
    return (
      <View
        style = { {
          flexDirection: "row",
          width: this.props.width - 40,
          height: this.props.width / 2 - 80,
          borderWidth: 0.5,
          borderColor: "#dddddd",
          marginBottom: 20,
          borderRadius: 3,
          borderWidth: 1,
          borderColor: "#dddddd",
          shadowOffset: { width: 0, height: 0 },
          shadowColor: "black",
          shadowOpacity: 0.1,
          elevation: 1,
          backgroundColor: "white"
        }}
      >
        <View style = { { flex: 2 } }>
          <Image
            style = { styles.image }
            source = { require("../../../assets/textbooks.jpg") }
          />
        </View>
        <View style = { styles.listingContainer }>
          <Text style = { styles.titleText }>
            { this.props.name }
          </Text>
          <Text style = { { alignItems: "center", fontSize: 12 } }>
            { this.props.category }
          </Text>
          <Text style = { { alignItems: "center", fontSize: 12 } }>
            ${this.props.price}
          </Text>
          <Text style = { { alignItems: "center", fontSize: 12 } }>
            seller: { this.props.seller }
          </Text>
        </View>
      </View>
    );
  }
}

export default Listings;

const styles = StyleSheet.create({
  image: {
    flex: 2,
    width: null,
    height: null,
    resizeMode: "cover",
    borderRadius: 3
  },
  listingContainer: {
    flex: 1,
    alignItems: "center",
    justifyContent: "space-evenly"
  },
  titleText: {
    alignItems: "center",
    fontSize: 14,
    fontWeight: "bold",
    justifyContent: "center",
    paddingHorizontal: 10
  }
});
