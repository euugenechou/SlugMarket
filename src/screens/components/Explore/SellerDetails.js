// React imports
import React from "react";
import {
  View,
  Text,
  StyleSheet,
  SafeAreaView,
  ScrollView,
  TouchableHighlight,
  Image,
  Dimensions,
  RefreshControl
} from "react-native";
// AWS imports
import { API } from "aws-amplify";
// Local imports
import UserListings from "../Profile/UserListings";

export default class SellerDetails extends React.Component {
  state = {
    postsToRender: [],
    refreshing: false
  };

  static navigationOptions = () => ({
    headerTintColor: "teal",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      shadowColor: "transparent",
      borderBottomWidth: 0
    }
  });

  componentWillMount() {
    this.getPostsToRender();
  }

  _onRefresh = () => {
    this.setState({ refreshing: true });
    this.getPostsToRender();
    this.setState({ refreshing: false });
  };

  getPostsToRender() {
    const path = "/itemPostings/allPosts";
    const apiName = "itemPostingsCRUD";
    const params = {
      headers: {},
      response: true
    };

    API.get(apiName, path, params)
      .then(response => {
        this.setState({
          postsToRender: response.data.filter(post => {
            console.log(post.userId);
            console.log(this.props.navigation.getParam("userId"));
            return (
              post.userId === this.props.navigation.getParam("userId") &&
              post.isSold === false
            );
          })
        });
      })
      .catch(err => console.log(err));
  }

  renderListings = () => {
    return this.state.postsToRender.map(post => {
      return (
        <TouchableHighlight
          onPress = { () =>
            this.props.navigation.navigate("ListingInfo", {
              itemName: post.itemName,
              price: post.price,
              seller: post.seller,
              category: post.category,
              description: post.description,
              timeAdded: post.timeAdded
            })
          }
          underlayColor="white"
          key = { post.timeAdded }
        >
          <UserListings
            width = { width }
            name = { post.itemName }
            price = { post.price }
            category = { post.category }
            seller = { post.seller }
          />
        </TouchableHighlight>
      );
    });
  };

  render() {
    return (
      <SafeAreaView style = { { backgroundColor: "white", flex: 1 } }>
        <ScrollView
          scrollEventThrottle = { 16 }
          showsVerticalScrollIndicator = { false }
          contentContainerStyle = { { backgroundColor: "white" } }
          refreshControl = {
            <RefreshControl
              refreshing = { this.state.refreshing }
              onRefresh = { this._onRefresh }
            />
          }
        >
          <View style = { { backgroundColor: "white" } }>
            <View style = { styles.topView }>
              <Image
                source = { require("../../../assets/darrell.png") }
                style = { styles.image }
              />
              <View style = { styles.midView } >
                <View style = { styles.botView }>
                  <Text style = { styles.titleText }>
                    { this.props.navigation.getParam("seller") }
                  </Text>
                  <Text style = { styles.text }> { this.props.navigation.getParam("email") } </Text>
                  <Text style = { styles.text }> { this.props.navigation.getParam("phoneNumber") } </Text>
                </View>
              </View>
            </View>
            <Text style = { styles.listingText }>
              { this.props.navigation.getParam("seller") }'s Listings
            </Text>
            { this.renderListings() }
          </View>
        </ScrollView>
      </SafeAreaView>
    );
  }
}

var { width, height } = Dimensions.get("window");

const styles = StyleSheet.create({
  topView: {
    backgroundColor: "white",
    flexDirection: "row-reverse",
    flex: 1
  },
  image: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: 40,
    width: null,
    height: 150,
    resizeMode: "contain",
    borderRadius: 8,
    flex: 1
  },
  midView: {
    backgroundColor: "white",
    flexDirection: "column",
    flex: 1,
    paddingTop: 60
  },
  botView: {
    alignItems: "center",
    backgroundColor: "white",
    justifyContent: "flex-start",
    paddingLeft: 20
  },
  text: {
    paddingTop: 5,
    fontSize: 14,
    fontWeight: "300",
    justifyContent: "center",
    alignItems: "center"
  },
  titleText: {
    fontSize: 36,
    fontWeight: "600",
    justifyContent: "center",
    alignItems: "center"
  },
  listingText: {
    fontSize: 24,
    fontWeight: "700",
    marginBottom: 20,
    marginTop: 20,
    paddingHorizontal: 20
  }
});
