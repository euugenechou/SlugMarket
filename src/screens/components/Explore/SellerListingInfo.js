// React imports
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";

export default class SellerListingInfo extends Component {
  static navigationOptions = () => ({
    headerTintColor: "white",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      shadowColor: "transparent",
      borderBottomWidth: 0
    },
    headerBackground: (
      <Image
        style = { styles.wrapper }
        source = { require("../../../assets/textbooks.jpg") }
      />
    ),
  });

  render() {
    return (
      <View style = { styles.container }>
        <Text style = { styles.titleText }>
          { this.props.navigation.getParam("itemName") }
        </Text>
        <Text style = { styles.text }>
          ${ this.props.navigation.getParam("price") }
        </Text>
        <Text style = { styles.text }>
          { this.props.navigation.getParam("category") }
        </Text>
        <Text style = { styles.text }>
          { this.props.navigation.getParam("description") }
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  text: {
    fontSize: 22,
    color: "black",
    fontWeight: "400",
    alignSelf: 'center',
    paddingBottom: 10,
    paddingHorizontal: 30,
  },
  titleText: {
    fontSize: 30,
    color: "black",
    fontWeight: "800",
    alignSelf: 'center',
    paddingBottom: 10,
    marginTop: 100
  },
  container: {
    flex: 1,
    marginTop: 20,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white"
  },
  buttonContainer: {
    width: 300,
    alignSelf: 'center',
    paddingTop: 10
  },
  wrapper: {
    ...StyleSheet.absoluteFillObject,
    marginTop: 20,
  },
});
