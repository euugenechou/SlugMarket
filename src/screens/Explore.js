/* React imports */
import React, { Component } from "react";
import {
  View,
  Text,
  SafeAreaView,
  TextInput,
  Platform,
  StatusBar,
  ScrollView,
  Dimensions,
  RefreshControl,
  TouchableHighlight
} from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import { Button } from "react-native-elements";
import { API } from "aws-amplify";

/* Local imports */
import Category from "./components/Explore/Category";
import Listings from "./components/Explore/Listings";
import { styles } from "../styles";

const { height, width } = Dimensions.get("window");

class Explore extends Component {
  state = {
    postsToRender: [],
    refreshing: false
  };

  static navigationOptions = () => ({
    header: null,
    headerStyle: {
      height: 40,
      shadowColor: "black",
      borderBottomColor: "black",
      borderBottomWidth: 0.5
    }
  });

  componentWillMount() {
    this.startHeaderHeight = 90;
    if (Platform.OS == "android") {
      this.startHeaderHeight = 100 + StatusBar.currentHeight;
    }
    this.getRecentPosts();
    console.log(this.state.postsToRender);
  }

  _onRefresh = () => {
    this.setState({ refreshing: true });
    this.getRecentPosts();
    this.setState({ refreshing: false });
  };

  /**
   * Gets items which were posted at most one day ago
   */
  getRecentPosts() {
    const path = "/itemPostings";
    const apiName = "itemPostingsCRUD";
    const headers = {
      headers: {},
      response: true,
      queryStringParameters: {
        ScanIndexForward: false
      }
    };
    API.get(apiName, path, headers)
      .then(response => {
        console.log(response);
        this.setState({
          postsToRender: response.data.sort((a, b) => {
            return new Date(b.timeAdded) - new Date(a.timeAdded);
          })
        });
      })
      .catch(error => console.log(error.response));
  }

  render() {
    return (
      <SafeAreaView style={{ flex: 1 }}>
        <View style={{ flex: 1, backgroundColor: "white" }}>
          <View
            style={{
              height: this.startHeaderHeight,
              backgroundColor: "white",
              borderBottomWidth: 0.5,
              borderBottomColor: "#dddddd",
              shadowOffset: { width: 0, height: 0 },
              shadowColor: "black",
              shadowOpacity: 0.1,
              elevation: 1
            }}
          >
            <View
              style={{
                flexDirection: "row",
                padding: 10,
                backgroundColor: "white",
                marginHorizontal: 20,
                shadowOffset: { width: 0, height: 0 },
                shadowColor: "black",
                shadowOpacity: 0.2,
                elevation: 1,
                marginTop: Platform.OS == "android" ? 30 : 30
              }}
            >
              <Icon name="ios-search" size={20} style={{ marginRight: 10 }} />
              <TextInput
                underlineColorAndroid="transparent"
                placeholder="Search"
                placeholderTextColor="grey"
                style={{
                  flex: 1,
                  fontWeight: "700",
                  backgroundColor: "white",
                  borderRadius: 3
                }}
              />
            </View>
          </View>
          <ScrollView
            scrollEventThrottle={16}
            showsVerticalScrollIndicator={false}
            refreshControl={
              <RefreshControl
                refreshing={this.state.refreshing}
                onRefresh={this._onRefresh}
              />
            }
          >
            <View style={{ flex: 1, backgroundColor: "white", paddingTop: 20 }}>
              <Text style={styles.sectionTitle}>Browse Items By Category</Text>
              <View style={styles.sideScroll}>
                <ScrollView
                  horizontal={true}
                  showsHorizontalScrollIndicator={false}
                >
                  <TouchableHighlight
                    onPress={() =>
                      this.props.navigation.navigate("ViewCategory", {
                        category: "textbook"
                      })
                    }
                    underlayColor="white"
                  >
                    <Category
                      imageUri={require("../assets/textbooks.jpg")}
                      name="Textbooks"
                    />
                  </TouchableHighlight>
                  <TouchableHighlight
                    onPress={() =>
                      this.props.navigation.navigate("ViewCategory", {
                        category: "furniture"
                      })
                    }
                    underlayColor="white"
                  >
                    <Category
                      imageUri={require("../assets/furniture.jpg")}
                      name="Furniture"
                    />
                  </TouchableHighlight>
                  <TouchableHighlight
                    onPress={() =>
                      this.props.navigation.navigate("ViewCategory", {
                        category: "electronic"
                      })
                    }
                    underlayColor="white"
                  >
                    <Category
                      imageUri={require("../assets/electronics.jpg")}
                      name="Electronics"
                    />
                  </TouchableHighlight>
                </ScrollView>
              </View>
              <View style={{ alignItems: "center", justifyContent: "center" }}>
                <Button
                  raised
                  color="white"
                  backgroundColor="teal"
                  borderRadius={5}
                  containerViewStyle={{ width: 300, paddingTop: 10 }}
                  onPress={() => this.props.navigation.navigate("AddItem")}
                  title="Sell An Item"
                  fontWeight="bold"
                />
              </View>
              <View style={{ marginTop: 20 }}>
                <Text
                  style={{
                    fontSize: 24,
                    fontWeight: "700",
                    paddingHorizontal: 20
                  }}
                >
                  Recently Added Items
                </Text>
                <View
                  style={{
                    paddingHorizontal: 20,
                    paddingVertical: 20,
                    marginBottom: 20,
                    flexDirection: "column",
                    flexWrap: "wrap",
                    justifyContent: "space-evenly"
                  }}
                >
                  {this.state.postsToRender.map(post => {
                    return (
                      <TouchableHighlight
                        onPress={() =>
                          this.props.navigation.navigate("ListingInfo", {
                            name: post.itemName,
                            price: post.price,
                            seller: post.seller,
                            category: post.category,
                            description: post.description
                          })
                        }
                        underlayColor="white"
                        key={post.timeAdded}
                      >
                        <Listings
                          width={width}
                          name={post.itemName}
                          price={post.price}
                          seller={post.seller}
                          category={post.category}
                        />
                      </TouchableHighlight>
                    );
                  })}
                </View>
              </View>
            </View>
          </ScrollView>
        </View>
      </SafeAreaView>
    );
  }
}

export default Explore;
