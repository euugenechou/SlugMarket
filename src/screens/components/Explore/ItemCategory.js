// React imports 
import React, { Component } from "react";
import {
  View,
  Text,
  StyleSheet,
  Image,
  TouchableHighlight,
  RefreshControl,
  ScrollView,
  Dimensions,
  SafeAreaView,
  Button
} from "react-native";

import Listings from "../Explore/Listings";
import { API } from "aws-amplify";

const { height, width } = Dimensions.get("window");

class ItemCategory extends Component {
  state = {
    postsToRender: [],
    category: this.props.navigation.getParam("category"),
    refreshing: false
  };

  static navigationOptions = () => ({
    headerTintColor: "teal",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      shadowColor: "transparent",
      borderBottomWidth: 0,
    }
  });

  componentWillMount() {
    this.getPostsByCategory();
    console.log(this.state.postsToRender);
  }

  _onRefresh = () => {
    this.setState({ refreshing: true });
    this.getPostsByCategory();
    this.setState({ refreshing: false });
  };

  getPostsByCategory() {
    const path = "/itemPostings/allPosts";
    const apiName = "itemPostingsCRUD";
    const headers = {
      headers: {},
      response: true,
      queryStringParameters: {
        ScanIndexForward: false
      }
    };
    API.get(apiName, path, headers)
      .then(response => {
        console.log(response);
        const sorted = response.data.sort((a, b) => {
          return new Date(b.timeAdded) - new Date(a.timeAdded);
        });
        this.setState({
          postsToRender: sorted.filter(post => {
            return (!post.isSold && 
            !post.isRemoved &&
            post.category === this.props.navigation.getParam("category"));
          })
        });
      })
      .catch(error => console.log(error.response));
  }

  render() {
    return (
      <SafeAreaView style={{ flex: 1 }}>
        <View style={{ flex: 1, backgroundColor: "white" }}>
          <ScrollView
            scrollEventThrottle={16}
            showsVerticalScrollIndicator={false}
            refreshControl={
              <RefreshControl
                refreshing={this.state.refreshing}
                onRefresh={this._onRefresh}
              />
            }
          >
            <View style={{ marginTop: 20 }}>
              <Text style={styles.titleText}>
                Recently Added To {this.props.navigation.getParam("category")}
              </Text>
              <View
                style={styles.listings}
              >
                {this.state.postsToRender.map(post => {
                  return (
                    <TouchableHighlight
                      onPress = { () =>
                        this.props.navigation.navigate("ListingInfo", {
                          name: post.itemName,
                          price: post.price,
                          seller: post.seller,
                          category: post.category,
                          description: post.description
                        })
                      }
                      underlayColor="white"
                      key = {post.timeAdded}
                    >
                      <Listings
                        width = { width }
                        name = { post.itemName }
                        price = { post.price }
                        seller = { post.seller }
                        category = { post.category }
                      />
                    </TouchableHighlight>
                  );
                })}
              </View>
            </View>
          </ScrollView>
        </View>
      </SafeAreaView>
    );
  }
}

export default ItemCategory;

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center"
  },
  listings: {
    paddingHorizontal: 20,
    paddingVertical: 20,
    marginBottom: 20,
    flexDirection: "column",
    flexWrap: "wrap",
    justifyContent: "space-evenly"
  },
  titleText: {
    fontSize: 22,
    fontWeight: "700",
    paddingHorizontal: 20
  }
});
