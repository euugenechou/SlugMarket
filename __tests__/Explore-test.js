// __tests__/Explore-test.js
import "react-native";
import React from "react";
import Explore from "../src/screens/Explore";

import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Explore />).toJSON();
  expect(tree).toMatchSnapshot();
});
