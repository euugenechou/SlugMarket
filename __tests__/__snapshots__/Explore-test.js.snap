// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "paddingTop": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "700",
              "paddingBottom": 30,
              "paddingHorizontal": 20,
            }
          }
        >
          Hi 
          
          ! 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "700",
              "paddingHorizontal": 20,
            }
          }
        >
          Browse Items By Category
        </Text>
        <View
          style={
            Object {
              "height": 130,
              "marginTop": 20,
            }
          }
        >
          <RCTScrollView
            horizontal={true}
            showsHorizontalScrollIndicator={false}
          >
            <View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#dddddd",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "elevation": 1,
                      "height": 120,
                      "marginLeft": 20,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "width": 130,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 2,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../src/assets/textbooks.jpg",
                        }
                      }
                      style={
                        Object {
                          "borderRadius": 3,
                          "borderWidth": 0.1,
                          "flex": 1,
                          "height": null,
                          "resizeMode": "cover",
                          "shadowColor": "black",
                          "shadowOffset": Object {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0.2,
                          "width": null,
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontWeight": "600",
                        }
                      }
                    >
                       
                      Textbooks
                       
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#dddddd",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "elevation": 1,
                      "height": 120,
                      "marginLeft": 20,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "width": 130,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 2,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../src/assets/furniture.jpg",
                        }
                      }
                      style={
                        Object {
                          "borderRadius": 3,
                          "borderWidth": 0.1,
                          "flex": 1,
                          "height": null,
                          "resizeMode": "cover",
                          "shadowColor": "black",
                          "shadowOffset": Object {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0.2,
                          "width": null,
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontWeight": "600",
                        }
                      }
                    >
                       
                      Furniture
                       
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#dddddd",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "elevation": 1,
                      "height": 120,
                      "marginLeft": 20,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "width": 130,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 2,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../src/assets/electronics.jpg",
                        }
                      }
                      style={
                        Object {
                          "borderRadius": 3,
                          "borderWidth": 0.1,
                          "flex": 1,
                          "height": null,
                          "resizeMode": "cover",
                          "shadowColor": "black",
                          "shadowOffset": Object {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0.2,
                          "width": null,
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontWeight": "600",
                        }
                      }
                    >
                       
                      Electronics
                       
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "700",
                "paddingHorizontal": 20,
              }
            }
          >
            Recently Added Items
          </Text>
          <View
            style={
              Object {
                "flexDirection": "column",
                "flexWrap": "wrap",
                "justifyContent": "space-evenly",
                "marginBottom": 20,
                "paddingHorizontal": 20,
                "paddingVertical": 20,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
